mono/metadata/boehm-gc.c

```diff
@@ -665,7 +654,7 @@ mono_gc_alloc_obj (MonoVTable *vtable, size_t size)
 		obj->vtable = vtable;
 	}
 
-	if (G_UNLIKELY (alloc_events))
+	if (G_UNLIKELY (mono_profiler_events & MONO_PROFILE_ALLOCATIONS))
```

Comment:
I think we should wrap this into a any macro that says whether a given profiler event it enabled. Including the G_UNLIKELY.